pbj-core/pbj-compiler/src/main/java/com/hedera/pbj/compiler/impl/Common.java

```diff
@@ -501,6 +504,210 @@ private static String getRepeatedEqualsGeneration(String generatedCodeSoFar, Fie
+							"""
+							result = Boolean.compare($fieldName, thatObj.$fieldName);
+							if (result != 0) {
+							    return result;
+							}
+							""".replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.FLOAT) {
+					generatedCodeSoFar +=
+							"""
+							result = Float.compare($fieldName, thatObj.$fieldName);
+							if (result != 0) {
+							    return result;
+							}
+							""".replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.DOUBLE) {
+					generatedCodeSoFar +=
+							"""
+							result = Double.compare($fieldName, thatObj.$fieldName);
+							if (result != 0) {
+							     return result;
+							}
+							""".replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.STRING ||
+						f.type() == Field.FieldType.ENUM) {
+					generatedCodeSoFar += generateCompareToForObject(f);
+				} else if (f.type() == Field.FieldType.BYTES) {
+					generatedCodeSoFar +=
+							"""
+							result = $fieldName.compareTo(thatObj.$fieldName);
+							if (result != 0) {
+								return result;
+							}
+							 """.replace("$fieldName", f.nameCamelFirstLower());
+				} else if (f.type() == Field.FieldType.MESSAGE || f.type() == Field.FieldType.ONE_OF) {
+					verifyComparable(f, destinationSrcDir);
+					generatedCodeSoFar += generateCompareToForObject(f);
+				}else {
+					throw new IllegalArgumentException("Unexpected field type for getting CompareTo - " + f.type().toString());
+				}
+			}
+		}
+		return generatedCodeSoFar.indent(DEFAULT_INDENT * 2);
+	}
+
+	@NonNull
+	private static String generateCompareToForObject(Field f) {
+		return """
+				if ($fieldName == null && thatObj.$fieldName != null) {
+				    return -1;
+				}
+				if ($fieldName != null && thatObj.$fieldName == null) {
+				    return 1;
+				}
+				if ($fieldName != null) {
+				    result = $fieldName.compareTo(thatObj.$fieldName);
+				}
+				if (result != 0) {
+				    return result;
+				}
+				""".replace("$fieldName", f.nameCamelFirstLower());
+	}
+
+	/**
+	 * Verify that the field is comparable.
+	 * @param field The field to verify.
+	 * @param destinationSrcDir The directory where the previously generated code is saved.
+	 */
+	private static void verifyComparable(final Field field, File destinationSrcDir) {
+		if (field instanceof final SingleField singleField) {
+			if (singleField.type() != Field.FieldType.MESSAGE) {
+				// everything else except message and bytes is comparable for sure
+				return;
+			}
+			// let's check if the message implements Comparable
+			final String className = singleField.javaFieldType();
+			final File javaFile = getJavaFile(destinationSrcDir, singleField.messageTypeModelPackage(), className);
+			try (BufferedReader reader = new BufferedReader(new FileReader(javaFile))) {
+				String line;
+				while ((line = reader.readLine()) != null) {
+					if (line.contains("implements Comparable")) {
```

Comment:
Just make sure nobody puts a comment to their .proto file with this exact substring! :)

Like mentioned above, I'd prefer to avoid this incomplete parsing method here if possible at all.

Reply:
Fair enough. I used `class\s+\w+\s+implements\s+Comparable` regex instead